### ISSUE TITLE
Simulation is working but image is static so agents and time are not moving.

### ISSUE THREAD

**Issue by Flanua** [SaturnCassini/gpt4all_generative_agents#7]
![222](https://github.com/SaturnCassini/gpt4all_generative_agents/assets/140022879/6cc0310c-72b8-4370-a753-9a6ee7dd7ed9)

------------------------------------
![111](https://github.com/SaturnCassini/gpt4all_generative_agents/assets/140022879/a8092dfa-26dd-43ac-934a-b3a44ca6be08)

------------------------------------
![44444](https://github.com/SaturnCassini/gpt4all_generative_agents/assets/140022879/8cf13e9e-7dae-43a3-9ef0-935a0560afed)

------------------------------------
![3333](https://github.com/SaturnCassini/gpt4all_generative_agents/assets/140022879/9c2816f5-1f71-47ce-943a-886de9a80442)

Actual simulation is working but from manage.py server console window all I see is 
a bunch of : [28/Aug/2023 17:59:59] "POST /update_environment/ HTTP/1.1" 200 14  lines when I open http://localhost:8000/simulator_home# and nothing is moving time is not progressing either agents status frozen too.
P.S so I just can't see what actually is happening except the backend server console window.
And I also tried to run replay of my simulation but nothing is working too only default replays are working.

My requirements.txt:

aiohttp==3.8.3
aiosignal==1.3.1
asgiref==3.5.2
async-generator==1.10
async-timeout==4.0.2
attrs==22.2.0
boto==2.49.0
botocore==1.29.43
certifi==2021.10.8
charset-normalizer==2.0.12
click==8.0.3
cycler==0.11.0
dj-database-url==0.5.0
Django==2.2
django-cors-headers==2.5.3
django-storages-redux==1.3.3
exceptiongroup==1.1.0
frozenlist==1.3.3
gensim==3.8.0
gunicorn==20.1.0
h11==0.14.0
idna==3.3
importlib-metadata==4.8.2
jmespath==1.0.1
joblib>=1.1.1
kiwisolver==1.4.4
matplotlib==3.7.2
multidict==6.0.4
nltk==3.6.5
numpy==1.25.2
gpt4all==1.0.5
outcome==1.2.0
packaging==23.0
pandas==2.0.3
patsy==0.5.3
Pillow==10.0.0
psycopg2-binary==2.9.5
pycparser==2.21
pyparsing==3.0.6
PySocks==1.7.1
python-dateutil==2.8.2
pytz==2021.3
regex==2023.8.8
requests==2.26.0
s3transfer==0.6.0
scikit-learn==1.3.0
scikit-posthocs==0.7.0
scipy==1.11.1
seaborn==0.12.2
selenium==4.8.2
six==1.16.0
sklearn==0.0
smart-open==5.2.1
sniffio==1.3.0
sortedcontainers==2.4.0
sqlparse==0.4.3
statsmodels==0.13.5
threadpoolctl==3.0.0
tqdm==4.62.3
trio==0.22.0
trio-websocket==0.9.2
trueskill==0.4.5
typing-extensions==4.0.0
urllib3==1.26.7
wsproto==1.2.0
yarl==1.8.2
yellowbrick==1.5
zipp==3.6.0
---------------------------------

And I also inserted missing addition code to create movement folder... but it doesn't seems to solve anything:

![5555](https://github.com/SaturnCassini/gpt4all_generative_agents/assets/140022879/23f4d27c-4083-4e60-987f-b8834942f222)

---------------------------------

### **So any fix or help would be nice to have. Myabe it's a problem with Django library who knows.**

**Comment by renzhangzhe** [comment]
I had the same problem, Have you solved it？

**Comment by Flanua** [comment]
> I had the same problem, Have you solved it？

Nope. :(

P.S: I also tried to create movement folder manually in case that's the problem (because that code that I inserted still doesn't creating that folder for some reason) and the new movement chunks (like 0,1,2,3,4,etc) in .json format in that folder are still not created automatically for my simulation and the simulation base: base_the_ville_isabella_maria_klaus so we can't see our agents and time moving we can only see it moving by playing saved by default replays because those replays like: July1_the_ville_isabella_maria_klaus-step-3-1 and etc have movement folder and those movement chunks in json format inside movement folder present.

**Comment by myrainbowandsky** [comment]
Same problem. any progress?

**Comment by YuehChuan** [comment]
same issue  python3.10  win11 with local model gpt4all

![image](https://github.com/SaturnCassini/gpt4all_generative_agents/assets/7314531/decaef63-bd97-473a-8570-7432e51db5de)

**Comment by YuehChuan** [comment]
put movement folder under
`generative_agents\environment\frontend_server\storage\<environment>`

run 2500

she is still sleeping.
https://github.com/joonspk-research/generative_agents/issues/44

![image](https://github.com/SaturnCassini/gpt4all_generative_agents/assets/7314531/ef340332-2d1e-4ad5-bcce-e19a26fef48e)

**Comment by YeeZ93** [comment]
Same problem, are there any updates? @Flanua @renzhangzhe @myrainbowandsky @YuehChuan @SaturnCassini 
Thx!

**Comment by sentball** [comment]
have any ideas? 
do your front work normal? @SaturnCassini

**Comment by SaturnCassini** [comment]
No my frontend wasnt working when i ran it, only in CLI, looking for PRs that can fix this

**Comment by Springbeann** [comment]
Was there any resolution to this? Currently experiencing something similar.

**Comment by DanielMarchand** [comment]
I think I have it working with mistralnemo (12GB) and will verify with llama3 in the future when I can. The problem is that all the prompts and their parsers had to be rewritten (which has been a lot of work). For the agents not moving issue its because the local models never generated output in the {format} requested by the code and so they kept resorting to the default which is not changing rooms. 

You can try out the branch that I have things working here: https://github.com/DanielMarchand/generative_agents/tree/fix-and-improve_add-ollama

Setup is a bit annoying leave an issue there if you have any questions. PS if i've pushed some backwards-incompatible changes use the 9d49a968e4e6df0c34205df95805cab6b82dec1f commit which I have just validated.
